Sources/AuthFoundation/OAuth2/OAuth2Client.swift

```diff
@@ -246,6 +246,10 @@ public class OAuth2Client {
                         case .failure(let error):
                             action.finish(.failure(.network(error: error)))
                             self.delegateCollection.invoke { $0.oauth(client: self, didRefresh: token, replacedWith: nil) }
+                            
+                            NotificationCenter.default.post(name: .tokenRefreshFailed,
+                                                            object: token,
+                                                            userInfo: ["error": error])
```

Comment:
would it make sense to move this note to the delegateCollection?

Reply:
The delegateCollection is used in a number of different areas, and I wouldn't want to add unnecessary overhead elsewhere.